tests/test_function_model.py

```diff
@@ -0,0 +1,2886 @@
+from inspect import Parameter, Signature, signature
+from sys import version_info
+from types import GenericAlias, NoneType
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+
+import pytest
+from pydantic_core import PydanticCustomError, PydanticKnownError, ValidationError
+
+from pydantic import TypeAdapter
+from pydantic.function_model import FunctionModel, get_annotation_type
+
+# Python 3.7 does not support positional-only arguments, which this uses for testing
+if version_info <= (3, 7):  # noqa: UP036
+    pytest.skip('These tests use positional-only parameters in functions, which are not supported prior to Python 3.8')
+
+
+"""def test_only_pos():  # noqa: C901
```

Comment:
What is the purpose of this string?

Reply:
These were just old tests that I had commented out while I was working on replacing them. Forgot to remove them, but have removed them now.